src/org/zaproxy/zap/extension/wappalyzer/WappalyzerPassiveScanner.java

```diff
@@ -96,17 +98,38 @@ public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
 						String body = msg.getResponseBody().toString();
 						for (AppPattern p : app.getHtml()) {
 							if (p.getPattern().matcher(body).find()) {
-								logger.debug("WAPP body matched " + app.getName());
+								LOGGER.debug("WAPP body matched " + app.getName());
 								found = true;
 								break;
 							}
 						}
 					}
 					if (! found) {
+						for (Map<String, AppPattern> sp : app.getMetas()) {

```

Comment:
This for loop (and the following) should be inside the following if statement instead (if there are no meta elements no need to loop the meta patterns).